src/controller/replication/execution.go

```diff
@@ -154,11 +154,11 @@ func (c *controller) Start(ctx context.Context, policy *replicationmodel.Policy,
 func (c *controller) markError(ctx context.Context, executionID int64, err error) {
 	logger := log.GetLogger(ctx)
 	// try to stop the execution first in case that some tasks are already created
-	if err := c.execMgr.StopAndWait(ctx, executionID, 10*time.Second); err != nil {
-		logger.Errorf("failed to stop the execution %d: %v", executionID, err)
+	if stopErr := c.execMgr.StopAndWait(ctx, executionID, 10*time.Second); stopErr != nil {
```

Comment:
Because the code change is execMgr, is it possible to extract a function and move it to execMgr?